packages/pdf-viewer/Page.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useRef, useState } from 'react';
+import useIsVisible from './hooks/useIsVisible';
+import { PdfData, ScaledSize } from './pdfSource';
+
+require('./pages.css');
+
+export default function Page(props: {
+	pdf: PdfData;
+	pageNo: number;
+	focusOnLoad: boolean;
+	isAnchored: boolean;
+	scaledSize: ScaledSize;
+	isDarkTheme: boolean;
+	container: React.MutableRefObject<HTMLElement>;
+}) {
+	const [error, setError] = useState(null);
+	const [page, setPage] = useState(null);
+	const [scale, setScale] = useState(null);
+	const [timestamp, setTimestamp] = useState(null);
+	const canvasEl = useRef<HTMLCanvasElement>(null);
+	const wrapperEl = useRef<HTMLDivElement>(null);
+	const isVisible = useIsVisible(canvasEl, props.container);
+
+	useEffect(() => {
+		const loader = async () => {
+			if (!isVisible || !page || !props.scaledSize || (scale && props.scaledSize.scale == scale)) return;
+			try {
+				const viewport = page.getViewport({ scale: props.scaledSize.scale || 1.0 });
+				const canvas = canvasEl.current;
+				canvas.width = viewport.width;
+				canvas.height = viewport.height;
+				const ctx = canvas.getContext('2d');
+				const pageTimestamp = new Date().getTime();
+				setTimestamp(pageTimestamp);
+				page.render({
+					canvasContext: ctx,
+					viewport,
+					// Used so that the page rendering is throttled to some extent.
+					// https://stackoverflow.com/questions/18069448/halting-pdf-js-page-rendering
+					continueCallback: function(cont: any) {
+						if (timestamp != pageTimestamp) {
+							return;
+						}
+						cont();
+					},
+				});
+				setScale(props.scaledSize.scale);
+
+			} catch (err) {
+				console.error('Error rendering page no.', props.pageNo, err);
+				setError(err);
+			}
+		};
+		loader()
+			.catch(console.error);
+	}, [page, props.scaledSize, isVisible]);
```

Comment:
Always pass the absolute necessary dependencies and nothing more. In this case, I think you only need `props.scaledSize.scale`. And `page.getViewport({ scale: props.scaledSize.scale || 1.0 })` I believe. Doing so prevents unnecessary rerenders.